homeassistant/components/light/lutron.py

```diff
@@ -0,0 +1,94 @@
+"""Support for Lutron lights."""
+import logging
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, Light
+from homeassistant.components.lutron import (
+    LutronDevice, LUTRON_DEVICES, LUTRON_GROUPS, LUTRON_CONTROLLER)
+from homeassistant.util.async import (
+    run_coroutine_threadsafe)
+
+DEPENDENCIES = ['lutron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Lutron lights."""
+    area_devs = {}
+    devs = []
+    for (area_name, device) in hass.data[LUTRON_DEVICES]['light']:
+        dev = LutronLight(hass, area_name, device,
+                          hass.data[LUTRON_CONTROLLER])
+        area_devs.setdefault(area_name, []).append(dev)
+        devs.append(dev)
+    add_devices(devs)
+
+    for area in area_devs:
+        if area not in hass.data[LUTRON_GROUPS]:
+            continue
+        grp = hass.data[LUTRON_GROUPS][area]
+        ids = list(grp.tracking) + [dev.entity_id for dev in area_devs[area]]
+        run_coroutine_threadsafe(
+            grp.async_update_tracked_entity_ids(ids), hass.loop).result()
+
+
+def to_lutron_level(level):
+    """Convert the given HASS light level (0-255) to Lutron (0.0-100.0)."""
+    return float((level * 100) / 255)
+
+
+def to_hass_level(level):
+    """Convert the given Lutron (0.0-100.0) light level to HASS (0-255)."""
+    return int((level * 255) / 100)
+
+
+class LutronLight(LutronDevice, Light):
+    """Representation of a Lutron Light, including dimmable."""
+
+    def __init__(self, hass, area_name, lutron_device, controller):
+        """Initialize the light."""
+        self._prev_brightness = None
+        LutronDevice.__init__(self, hass, area_name, lutron_device, controller)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        new_brightness = to_hass_level(self._lutron_device.last_level())
```

Comment:
in the previous PR, @balloob  asked if this is doing I/O... It does not. last_level() just returns a cached value.